cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -700,3 +709,13 @@ func (scaleSet *ScaleSet) invalidateInstanceCache() {
 	scaleSet.lastInstanceRefresh = time.Now().Add(-1 * vmssInstancesRefreshPeriod)
 	scaleSet.instanceMutex.Unlock()
 }
+
+func (scaleSet *ScaleSet) invalidateSizeCacheWithLock() {
+	scaleSet.sizeMutex.Lock()
+	scaleSet.lastSizeRefresh = time.Now().Add(-1 * scaleSet.sizeRefreshPeriod)
+	scaleSet.sizeMutex.Unlock()
+
+	scaleSetStatusCache.mutex.Lock()
```

Comment:
Pointing this diff out:

I don't think the initial intention was working as expected, from the comment:

```
// Invalidate the VMSS size cache in order to fetch the size from the API.
```

However, there's another safeguard in `getVMSSInfo` that would also read data from cache if `scaleSet.sizeRefreshPeriod` hasn't passed from the last `scaleSetStatusCache` refresh.

Reply:
Right, this is not caught in our unit tests.  Could you help to add a unit test for the cache refreshing?

Reply:
yup, i'll look into adding one.